jupyter/analysis-automation/BasicExample.ipynb

```diff
@@ -0,0 +1,271 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import asyncio\n",
+    "from systemlink.clients.nitdmreader.api.metadata_api import MetadataApi\n",
+    "from systemlink.clients.nitdmreader.models.file_list import FileList\n",
+    "\n",
+    "metadata_api = MetadataApi()\n",
+    "import scrapbook as sb\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "from systemlink.clients.nitdmreader import *\n",
+    "data_api = DataApi()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# --- Metadata helper methods ---\n",
+    "\n",
+    "def get_groups(file_info):\n",
+    "    if not hasattr(file_info, '__getitem__'):\n",
```

Comment:
I guess this code was introduced to handle a bug in the TDM Reader API which no longer exists, so we should remove all of this helper methods and simply use `for group in file_info.groups` to iterate over groups.